packages/frontend/core/src/desktop/pages/workspace/attachment/index.tsx

```diff
@@ -0,0 +1,134 @@
+import { Skeleton } from '@affine/component';
+import {
+  type AttachmentBlockModel,
+  matchFlavours,
+} from '@blocksuite/affine/blocks';
+import { type Doc, DocsService, useService } from '@toeverything/infra';
+import { type ReactElement, useLayoutEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { AttachmentViewerView } from '../../../../components/attachment-viewer';
+import { ViewIcon, ViewTitle } from '../../../../modules/workbench';
+import { PageNotFound } from '../../404';
+import * as styles from './index.css';
+
+enum State {
+  Loading,
+  NotFound,
+  Found,
+}
+
+const useLoadAttachment = () => {
+  const { pageId, attachmentId } = useParams();
+  const docsService = useService(DocsService);
+  const [doc, setDoc] = useState<Doc | null>(null);
+  const [state, setState] = useState(State.Loading);
+  const [model, setModel] = useState<AttachmentBlockModel | null>(null);
+
```

Comment:
doc may not exist, so it is better to have a separate effect that waits for doc to appear.

```
const docRecord = useLiveData(docsService.list.doc$(docId))
const [doc, setDoc] = useState<Doc | null>(null);
useLayoutEffect(() => {
    if (!docRecord) {
      return;
    }
    const { doc, release } = docsService.open(pageId);
    setDoc(doc);
    return () => {
      release();
    };
  }, [...]);
```